File: Unique Binary Search Trees/Unique Binary Search Trees.py
class Solution:
    # @return an integer
    def numTrees(self, n):
        if(n == 0):
            return 1
        elif(n == 1):
            return 1
        else:
            result = 0
            for i in range(n):
                result += self.numTrees(i) * self.numTrees(n - i - 1)
            return result
        

File: Unique Binary Search Trees/Unique Binary Search Trees.java
public class Solution {
    public int numTrees(int n) {
        if(n == 0 || n == 1){
            return 1;
        }
        else{
            int res = 0;
            for(int i = 0; i < n; ++i){
                res += numTrees(i) * numTrees(n - i - 1);
            }
            return res;
        }
    }
}
